src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -101,7 +101,7 @@ public MappedLocalFunction(SynthesizedLambdaMethod symbol, ClosureKind closureKi
         // True if the rewritten tree should include assignments of the
         // original locals to the lifted proxies. This is only useful for the
         // expression evaluator where the original locals are left as is.
-        private readonly bool _assignLocals;
+        private readonly HashSet<LocalSymbol> _assignLocals;
```

Comment:
>_assignLocals [](start = 46, length = 13)

If this is null then all locals are assigned, otherwise only those that are in this set?